### DISCUSSION TITLE
How to connect from multiple processes to the same db file

### DISCUSSION THREAD

**Discussion by acirtep** [duckdb/duckdb#5946] [Q&A] [answered]
Hi all,

I have created a duckdb file, persisted on my local machine. I am trying to connect from 2 different Python shells to the same file, but I get `resource unavailable`. The aim is to be able to read from other processes, while only one is writing.

The steps that I followed are:

1. Created the .duckdb file from CLI with `.open .../mydatabase.duckdb` 
2. Created a table in the CLI and closed the connection (by exiting the shell)
3. Connect in one Python shell with `duckdb.connect(database='.../mydatabase.duckdb', read_only=True)`
4. Connect in a second Python shell with `duckdb.connect(database='.../mydatabase.duckdb', read_only=False)`

Step 4 fails with 
```In [2]: duckdb_conn = duckdb.connect(database='.../mydatabase.duckdb', read_only=False)
---------------------------------------------------------------------------
IOException                               Traceback (most recent call last)
Cell In[2], line 1
----> 1 duckdb_conn = duckdb.connect(database='.../mydatabase.duckdb', read_only=False)

IOException: IO Error: Could not set lock on file ".../mydatabase.duckdb": Resource temporarily unavailable
```

At the same time if I open a new CLI connection with `.open READ_ONLY '.../mydatabase.duckdb'` , the connection is successful, but the table created at step 2 is not recognized:
```D .open READ_ONLY '.../mydatabase.duckdb' 
D select * from duckdb_tables();
┌─────────────┬────────────┬────────────┬───────────┬──────────┬───────────┬───┬────────────────┬──────────────┬─────────────┬──────────────────────┬─────────┐
│ schema_name │ schema_oid │ table_name │ table_oid │ internal │ temporary │ … │ estimated_size │ column_count │ index_count │ check_constraint_c…  │   sql   │
│   varchar   │   int64    │  varchar   │   int64   │ boolean  │  boolean  │   │     int64      │    int64     │    int64    │        int64         │ varchar │
├─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│                                                                           0 rows                                                                            │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
D 
```
while in the Python shell opened in step 3, I get:
```In [12]: duckdb_conn.execute("select * from duckdb_tables()")
Out[12]: <duckdb.DuckDBPyConnection at 0x1053b5c30>

In [13]: duckdb_conn.df()
Out[13]: 
  schema_name  schema_oid              table_name  table_oid  ...  column_count  index_count  check_constraint_count                                                sql
0        main           1  electricity_daily_load        831  ...            27            0                       0  CREATE TABLE electricity_daily_load("Time" TIM...

[1 rows x 12 columns]

In [14]: 
```
My understanding is that I should be able to open as many read only connections (through CLI or other clients) as I wish. Is there an issue with the fact that I have created the database file from the CLI with the `.open` command?

Thank you,
Petra

**Comment by Mytherin** [comment, marked as answer]
Multiple processes connecting to the same database file is only possible if they are all read-only connections. It is not possible to have a read-write connection and a read-only connection connected to the same database file across multiple processes. See https://github.com/duckdb/duckdb/issues/77 for more background.
